web/static/js/components/respondents/RespondentsFilter.jsx

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { Component } from 'react'
+import { translate } from 'react-i18next'
+import AwesomeDebouncePromise from 'awesome-debounce-promise'
+
+type Props = {
+  t: Function,
+  inputValue: string,
+  onChange: Function,
+  onApplyFilter: Function
+}
+
+type State = {
+  inputValue: string,
+  applyFilterDebounced: Function
+}
+
+class RespondentsFilter extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      inputValue: props.inputValue,
+      applyFilterDebounced: AwesomeDebouncePromise(
+        props.onApplyFilter,
+        500
+      )
+    }
+  }
+
+  onInputChange = (event) => {
+    const { onChange } = this.props
+    const { applyFilterDebounced } = this.state
+    const inputValue = event.target.value
+    this.setState({ inputValue: inputValue })
+    onChange(inputValue)
+    applyFilterDebounced()
+  }
+
+  onInputKeyPress = (event) => {
+    const { onApplyFilter } = this.props
+    if (event.key === 'Enter') onApplyFilter()
```

Comment:
This should cancel existing de promise execution. Otherwise a later request will fire.